src/frontend/src/optimizer/plan_node/logical_multi_join.rs

```diff
@@ -483,6 +484,208 @@ impl LogicalMultiJoin {
         Ok(join_ordering)
     }
 
+    pub fn as_bushy_tree_join(&self) -> Result<PlanRef> {
+        // Join tree internal representation
+        #[derive(Clone, Default, Debug)]
+        struct JoinTreeNode {
+            idx: Option<usize>,
+            left: Option<Box<JoinTreeNode>>,
+            right: Option<Box<JoinTreeNode>>,
+            height: usize,
+        }
+
+        // join graph internal representation
+        #[derive(Clone, Debug)]
+        struct GraphNode {
+            id: usize,
+            join_tree: JoinTreeNode,
+            relations: HashSet<usize>,
+        }
+
+        let mut nodes: HashMap<_, _> = (0..self.inputs.len())
+            .map(|idx| GraphNode {
+                id: idx,
+                relations: HashSet::new(),
+                join_tree: JoinTreeNode {
+                    idx: Some(idx),
+                    left: None,
+                    right: None,
+                    height: 0,
+                },
+            })
+            .enumerate()
+            .collect();
+        let (eq_join_conditions, _) = self
+            .on
+            .clone()
+            .split_by_input_col_nums(&self.input_col_nums(), true);
+
+        for ((src, dst), _) in eq_join_conditions {
```

Comment:
Only using the existing join conditions is not enough, we need to do some derivation to gain more equal conditions. For example: `a == b && a == c` => `b == c`

Reply:
I added the derived conditions manually and in this way, the algorithm can generate the optimal bushy tree ordering.